async-stream/src/yielder.rs

```diff
@@ -87,7 +87,7 @@ impl<T> Receiver<T> {
 
 // ===== impl Enter =====
 
-impl<'a, T> Drop for Enter<'a, T> {
+impl<T> Drop for Enter<'_, T> {
```

Comment:
I disagree with the warning in this case. :(

Reply:
Well, this is what is also covered in single_use_lifetimes (which is allowed by default), so maybe clippy::needless_lifetimes should not warn it.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=351c8940c3d6dd7f311c7a3552280daf

```
warning: lifetime parameter `'a` only used once
 --> src/lib.rs:7:6
  |
7 | impl<'a, T> Drop for Enter<'a, T> {
  |      ^^ this lifetime...   -- ...is used only here
  |
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![warn(single_use_lifetimes)]
  |         ^^^^^^^^^^^^^^^^^^^^
help: elide the single-use lifetime
  |
7 - impl<'a, T> Drop for Enter<'a, T> {
7 + impl<T> Drop for Enter<'_, T> {
  |
```